### PR TITLE
rgw : catch non int exception

### PR DESCRIPTION
**Pull request by BryceCao**
Fixes: https://tracker.ceph.com/issues/48034
Signed-off-by: caolei halei15848934852@163.com

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by BryceCao** [comment]
@batrick , sorry, i close last PR by mistake, please label it again.
@pponnuvel , thank you for the suggestion, i have modified my PR.

File: src/rgw/rgw_auth.cc [hunk 1]
```diff
@@ -323,10 +323,22 @@ rgw::auth::Strategy::apply(const DoutPrefixProvider *dpp, const rgw::auth::Strat
     } catch (const int err) {
       ldpp_dout(dpp, 5) << "applier throwed err=" << err << dendl;
       return err;
+    } catch (const std::exception& e) {
+      ldpp_dout(dpp, 5) << "applier throwed unexpected err: " << e.what()
+                        << dendl;
+      return -EPERM;
+    } catch (...) {
```
Comment (dang) [inline src/rgw/rgw_auth.cc:330]: The ... catch shouldn't be necessary, since std::exception should catch everything, right?

Comment (pponnuvel) [reply src/rgw/rgw_auth.cc:330]: Shouldn't be in general unless there's some exception class that doesn't inherit from std::exception (I don't know if there's any such class involved here).

Comment (BryceCao) [reply src/rgw/rgw_auth.cc]: I checked the code here again. Actually, there would be two kinds of errors: int and std::exception, so i remove the part of `catch (...)`.

**Comment by BryceCao** [comment]
jenkins test make check

**Review by pponnuvel — APPROVED** [review]
LGTM

**Review by dang — APPROVED** [review]

**Comment by dang** [comment]
@cbodley do we think this needs Teuthology?

**Comment by harishmunjulur** [comment]
https://trello.com/c/lKg1n6nh, https://pulpito.ceph.com/hmunjulu-2020-11-20_22:05:31-rgw-wip-hmunjulu01-testing-2020-11-20-1051-distro-basic-smithi/

**Pull request merged**